packages/router/src/provide_router.ts

```diff
@@ -621,6 +621,60 @@ export function withHashLocation(): RouterConfigurationFeature {
   return routerFeature(RouterFeatureKind.RouterConfigurationFeature, providers);
 }
 
+/**
+ * A type alias for providers returned by `withHashLocation` for use with `provideRouter`.
+ *
+ * @see `withHashLocation`
+ * @see `provideRouter`
+ *
+ * @publicApi
+ */
+export type ErrorHandlerFeature = RouterFeature<RouterFeatureKind.ErrorHandlerFeature>;
+
+/**
+ * Subscribes to the Router's navigation events and calls the given function when a
+ * `NavigationError` happens.
+ *
+ * This function is run inside application's injection context so you can use the `inject` function.
+ *
+ * @usageNotes
+ *
+ * Basic example of how you can use the error handler option:
+ * ```
+ * const appRoutes: Routes = [];
+ * bootstrapApplication(AppComponent,
+ *   {
+ *     providers: [
+ *       provideRouter(appRoutes, withErrorHandler((e) => inject(MyErrorTracker).trackError(e))
```

Comment:
nit: it might be beneficial to add a type here, so that it's clear that an error has a special type.
```suggestion
 *       provideRouter(appRoutes, withErrorHandler((e: NavigationError) => inject(MyErrorTracker).trackError(e))
```